packages/app-mobile/components/screens/ConfigScreen/plugins/SearchPlugins.tsx

```diff
@@ -133,26 +131,30 @@ const PluginSearch: React.FC<Props> = props => {
 		}
 	}, [onInstall, props.themeId, props.pluginSettings, props.updatingPluginIds, props.updatablePluginIds, props.onShowPluginInfo, props.callbacks]);
 
-	const renderResultsCount = () => {
-		if (!searchQuery.length) return null;
+	const onClearSearch = useCallback(() => {
+		setSearchQuery('');
+	}, [setSearchQuery]);
 
-		return <Text style={styles.resultsCounter} variant='labelLarge'>
-			{_('Results (%d):', searchResults.length)}
-		</Text>;
+	const renderSearchButton = () => {
+		if (searchQuery) {
+			return <TextInput.Icon onPress={onClearSearch} accessibilityLabel={_('Clear search')} icon='close' />;
+		} else {
+			return <TextInput.Icon icon='magnify' aria-hidden={true} importantForAccessibility='no-hide-descendants'/>;
+		}
 	};
 
 	return (
 		<View style={styles.container}>
 			<TextInput
 				testID='searchbar'
 				mode='outlined'
-				left={<TextInput.Icon icon='magnify' />}
-				placeholder={_('Search plugins')}
+				right={renderSearchButton()}
+				placeholder={_('Search for plugins...')}
 				onChangeText={setSearchQuery}
 				value={searchQuery}
 				editable={props.repoApiInitialized}
 			/>
-			{renderResultsCount()}
+			<SectionLabel visible={!!searchQuery.length}>{_('Results (%d):', searchResults.length)}</SectionLabel>
 			<FlatList
```

Comment:
This `FlatList` is currently somewhat slow when there is a large number of search results. Making the plugin cards have a fixed height and providing a [getItemLayout](https://reactnative.dev/docs/flatlist#getitemlayout) could help with performance. If this is done, it should be in a follow-up pull request.

Reply:
- [ ] To-do: Create issue for this.